### PR TITLE
[DT-040319] Microsoft Exchange Office 365 Module

### PR DESCRIPTION
**Pull request by deci-shawns**
Added flexibility to list multiple pages of emails when listing or searching

### CODE REVIEW

File: Decisions.Microsoft365.Exchange/GraphRest.cs [hunk 1]
```diff
@@ -11,7 +11,7 @@ public class GraphRest
     {
         public static HttpResponseMessage HttpResponsePost(ExchangeSettings? settingsOverride, string urlExtension, HttpContent content)
         {
-            return SendHttpRequest(settingsOverride, urlExtension, content, HttpMethod.Post);
+            return CreateHttpRequest(settingsOverride, urlExtension, content, HttpMethod.Post);
```
Comment (jason-decisions) [inline Decisions.Microsoft365.Exchange/GraphRest.cs:14]: What is the purpose of these changes  vs. just modifying the parameters?

File: Decisions.Microsoft365.Exchange/GraphRest.cs [hunk 2]
```diff
@@ -90,6 +89,12 @@ public static HttpResponseMessage Delete(ExchangeSettings? settingsOverride, str
                 throw new BusinessRuleException("The request was unsuccessful.", ex);
             }
         }
+        
+        private static HttpResponseMessage CreateHttpRequest(ExchangeSettings? settingsOverride, string urlExtension,
+            HttpContent? content, HttpMethod httpMethod)
+        {
+            return SendHttpRequest(settingsOverride, $"{GetSettings(settingsOverride).GraphUrl}{urlExtension}", content, httpMethod);
+        }
```
Comment (jason-decisions) [inline Decisions.Microsoft365.Exchange/GraphRest.cs:97]: Why the need for a new method?

File: Decisions.Microsoft365.Exchange/GraphRest.cs (hunk 1 above)
Comment (deci-shawns) [reply Decisions.Microsoft365.Exchange/GraphRest.cs:14]: CreateHttpRequest will take the URL extension input and combine it with the settings/override base URL, which is how all of the steps are passing it along. See below for further explanation.

File: Decisions.Microsoft365.Exchange/GraphRest.cs (hunk 2 above)
Comment (deci-shawns) [reply Decisions.Microsoft365.Exchange/GraphRest.cs:97]: SendHttpRequest takes a URL input that is fully configured by that point: so it can accept the BaseURL+Extension that each step sends while also allowing the OData URLs to request data.

File: Decisions.Microsoft365.Exchange/GraphRest.cs (hunk 1 above)
Comment (jason-decisions) [reply Decisions.Microsoft365.Exchange/GraphRest.cs:14]: Right, but can't you do that directly when you send the parameter to SendHttpRequest?

`return SendHttpRequest(settingsOverride, $"{GetSettings(settingsOverride).GraphUrl}{urlExtension}", content, HttpMethod.Post)`

File: Decisions.Microsoft365.Exchange/GraphRest.cs (hunk 2 above)
Comment (jason-decisions) [reply Decisions.Microsoft365.Exchange/GraphRest.cs:97]: See my comment above.

File: Decisions.Microsoft365.Exchange/GraphRest.cs (hunk 1 above)
Comment (deci-shawns) [reply Decisions.Microsoft365.Exchange/GraphRest.cs:14]: Yes, but it seemed redundant to retrieve the settings within each step

Comment (jason-decisions) [reply Decisions.Microsoft365.Exchange/GraphRest.cs:14]: You're still retrieving it every time you're just doing it in a different method. I guess it's fine.

**Review by jason-decisions — APPROVED** [review]
